pymc_marketing/mmm/marginal_effects.py

```diff
@@ -0,0 +1,173 @@
+            X_new = self.create_intervention(sweep_value)
+            counterfac = self.mmm.predict(X_new, extend_idata=False, progressbar=False)
+            actual = self.mmm._get_group_predictive_data(
+                group="posterior_predictive", original_scale=True
+            )["y"]
+            uplift = counterfac - actual
+            predictions.append(uplift)
+
+        self.results = (
+            xr.concat(predictions, dim="sweep")
+            .assign_coords(sweep=self.sweep_values)
+            .transpose(..., "sweep")
+        )
+
+    def create_intervention(self, sweep_value):
+        """Apply the intervention to the predictors."""
+        X_new = self.X.copy()
+        if self.sweep_type == "multiplicative":
+            for predictor in self.predictors:
+                X_new[predictor] *= sweep_value
+        elif self.sweep_type == "additive":
+            for predictor in self.predictors:
+                X_new[predictor] += sweep_value
+        elif self.sweep_type == "absolute":
+            for predictor in self.predictors:
+                X_new[predictor] = sweep_value
+        else:
+            raise ValueError(f"Unsupported sweep_type: {self.sweep_type}")
+        return X_new
+
+    def plot_uplift(self, hdi_prob: float = 0.94, ax=None) -> plt.Axes:
+        """Plot the counterfactual uplift curve."""
+        if ax is None:
+            fig, ax = plt.subplots(figsize=(10, 6))
+
+        x = self.sweep_values
+        y = self.results.mean(dim=["chain", "draw"]).sum(dim="date")
+        ax.plot(x, y, label="Posterior mean", color="C0")
+
+        az.plot_hdi(
+            x,
+            self.results.sum(dim="date"),
+            hdi_prob=hdi_prob,
+            color="C0",
+            fill_kwargs={"alpha": 0.5, "label": f"{hdi_prob * 100:.0f}% HDI"},
+            plot_kwargs={"color": "C0", "alpha": 0.5},
+            smooth=False,
+            ax=ax,
+        )
+
+        ax.set(title="Counterfactual uplift plot")
+        if self.sweep_type == "absolute":
+            ax.set_xlabel(f"Absolute value of: {self.predictors}")
+        else:
+            ax.set_xlabel(
+                f"{self.sweep_type.capitalize()} change of: {self.predictors}"
+            )
+        ax.set_ylabel("Total uplift (sum over dates)")
+        plt.legend()
+        return ax
+
+    def compute_marginal_effects(self):
+        """Compute marginal effects via finite differences from the sweep results."""
+        sweep_axis = self.results.get_axis_num("sweep")
+        marginal_effects = np.gradient(
+            self.results, self.sweep_values, axis=sweep_axis, edge_order=2
+        )
+        self.marginal_effects = xr.DataArray(
+            marginal_effects,
+            dims=self.results.dims,
+            coords=self.results.coords,
+        )
+        return self.marginal_effects
+
+    def plot_marginal_effects(self, hdi_prob: float = 0.94, ax=None) -> plt.Axes:
+        """Plot the marginal effects curve."""
```

Comment:
Add the Parameters and Returns in docstring

Reply:
done in https://github.com/pymc-labs/pymc-marketing/pull/1673/commits/00e43099e81cd4221e7ab89e1ce53888516938bf